snuba/utils/streams/kafka.py

```diff
@@ -658,36 +658,30 @@ def __init__(
         self.__commit_log_topic = commit_log_topic
         self.__group_id = configuration["group.id"]
 
-    def poll(self, timeout: Optional[float] = None) -> Optional[Message[TPayload]]:
-        self.__producer.poll(0.0)
-        return super().poll(timeout)
-
     def __commit_message_delivery_callback(
-        self, error: Optional[KafkaError], message: ConfluentMessage
+        self, commit: Commit, future: Future[Commit]
     ) -> None:
-        if error is not None:
-            raise Exception(error.str())
+        try:
+            message = future.result()
+        except Exception:
+            logger.warning("Failed to produce commit record: %r", commit, exc_info=True)
+        else:
+            logger.debug("Produced commit record: %r", message)
```

Comment:
Slight behavior change here, this callback is now fired in the producer worker thread (see the logs below, the field after the timestamp is the thread ID) so a produce failure here will not cause this consumer to fail-stop. I think that should be fine — this is logged, and consumer lag on any consumers that depend on the commit log should be monitored, and whenever the consumer is able to report its offsets at a later time, progress will resume — but it is different.